draft-ietf-rats-eat.md

```diff
@@ -1102,94 +1102,46 @@ The values for the results enumerated type are as follows:
-If it was conveyed in parallel with the surrounding token, there would be no such binding and attackers could substitute a good attestation from another device for the attestation of an errant subsystem.
-
-A nested token does not need to use the same encoding as the enclosing token.
-This is to allow composite devices to be built without regards to the encoding supported by their attesters.
-Thus, a CBOR-encoded token like a CWT can have a JWT as a nested token submodule and vice versa.
-
-
-###### Surrounding EAT is CBOR-Encoded
-
-This describes the encoding and decoding of CBOR or JSON-encoded tokens nested inside a CBOR-encoded token.
-
-If the nested token is CBOR-encoded, then it MUST be a CBOR tag and MUST be wrapped in a byte string.
-The tag identifies whether the nested token is a CWT, a CBOR-encoded detached EAT bundle, or some other CBOR-format token defined in the future.
-A nested CBOR-encoded token that is not a CBOR tag is NOT allowed.
-
-If the nested token is JSON-encoded, then the data item MUST be a text string containing JSON.
-The JSON is defined in CDDL by JSON-Nested-Token in the next section.
-
-When decoding, if a byte string is encountered, it is known to be a nested CBOR-encoded token.
-The byte string wrapping is removed.
-The type of the token is determined by the CBOR tag.
-
-When decoding, if a text string is encountered, it is known to be a JSON-encoded token.
-The two-item array is decoded and tells the type of the JSON-encoded token.
-
 ~~~~CDDL
 {::include nc-cddl/nested-token-cbor.cddl}
 ~~~~
+~~~~CDDL
+{::include nc-cddl/nested-token-json.cddl}
+~~~~
+~~~~CDDL
+{::include nc-cddl/detached-digest.cddl}
+~~~~
 
-###### Surrounding EAT is JSON-Encoded
-
-This describes the encoding and decoding of CBOR or JSON-encoded tokens nested inside a JSON-encoded token.
+When decoding a submodule in a CBOR-encoded EAT, the Claims-Set type will be encoded as a map, the Nested-Token type as a CBOR-tagged object and the Detached-Submodule-Digest type as an array.
 
-The nested token MUST be an array of two, a text string type indicator and the actual token.
+When decoding a JSON-encoded EAT, a little more work is required because both the Nested-Token and Detached-Submodule-Digest types are arrays.
+To distinguish the nested token from the detached digest, the first element in the array is examined.
+If it is "JWT" or "BUNDLE", then the submodule is a Nested-Token.
+Otherwise it will contain an algorithm identifier and is a Detached-Submodule-Digest.
 
-The string identifying the JSON-encoded token MUST be one of the following:
+Where Nested-Token is used, nested CBOR EATs MUST be tagged, i.e., a CBOR tag will be used to distinguish between CWT or BUNDLE. Nested JSON EATs will be encoded as an JSON-Nested-Token, with the type indicated in the type field, i.e., first element in the array. The string identifying the JSON-encoded token MUST be one of the following:
```

Comment:
If you ask Carsten a CBOR tag is the combination of the tag number and the tag content. I thus never use the word "tagged" and say that the thing "is a tag". See my original text.

Very confusing use of the word "tag" IMO.

Reply:
You lost me on this one. There are quite a few *types* that feature the word tagged. The term "CBOR tag" appears in several places in prior versions including in a section title. The text here bears much in common with the language in 4.2.19.1.2.1 to my eye and section 4.2.19.1.2.2 features the phrase "is a tag". 

Tags play a central role in CBOR encodings in this spec. I don't see any way to avoid the term or reason to avoid the term (yet, at least). I made no change pending further clarification.

Reply:
Let me try again. A CBOR "tag" is NOT the tag number. It is the tag number AND the tag content. That's like saying a tag is not an indicator/label/identifier you put on a box but is instead the indicator plus the box. This is a very weird use of the term, but it is what it is.

It took me a long time to get clear on this especially as some published RFCs don't use the word this way and misled me. RFC 7949 and early drafts of 8949 where unclear and even wrong.

To me this makes use of the word "tagging" and "tagged" undesirable to me (even though section 3.4 in RFC 8949 uses them). I have been trying to write that something "is a tag" rather than something "is tagged".

Reply:
That helped. Thanks. I changed "CBOR EATs MUST be tagged" to "CBOR EATs MUST be a tag". I think that gets at the point.